### ISSUE TITLE
Rule `indent` does not warn on multi-line condition in es6

### ISSUE THREAD

**Issue by eTorAken** [eslint/eslint#2308]
This code (es6) is not considered a warning by the `indent` rule, but according to me it should! Here is the code:
```js
(() => myNumber.should.be.a.Number &&
       myString.should.be.a.String)();
```

This should be (with 2 spaces indent):
```js
(() => myNumber.should.be.a.Number &&
  myString.should.be.a.String)();
```

**Comment by lo1tuma** [comment]
Maybe related to #1801.

**Comment by nzakas** [comment]
This one will be a bit tricky as I imagine people will want different indent treatment in this case.

**Comment by alberto** [comment]
Closing by lack of activity
